### PR TITLE
v0.1

### PR DESCRIPTION
**Pull request by Darkstone2**
Added JSONBuilder Class

### CODE REVIEW

File: JsonBuilder/JsonBuilder/JsonBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,225 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Dynamic;
+using System.Linq;
+using System.Text;
+using System.Web.Script.Serialization;
+
+namespace SettingSaver
```
Comment (hishamco) [inline JsonBuilder/JsonBuilder/JsonBuilder.cs]: Change the namespace to `JsonBuilder`

File: JsonBuilder/JsonBuilder/JsonBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+{
+    public enum Formatting
```
Comment (hishamco) [inline JsonBuilder/JsonBuilder/JsonBuilder.cs]: Move the `enum` to a separated file

File: JsonBuilder/JsonBuilder/JsonBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+        /// <param name="jsonString">JSON string</param>
+        /// <returns>The property value your require. example: ToObject(Person).HisCourses</returns>
+        public static Dictionary<string, dynamic> ToObject(string jsonString)
+        {
+            if (!IsValidJson(jsonString))
+                throw new Exception("Not Valid Json Object!");
+
+            var jsDes = JavaScriptSerializer.Deserialize<dynamic>(jsonString);
+            return jsDes;
+            #region MyRegion
```
Comment (hishamco) [inline JsonBuilder/JsonBuilder/JsonBuilder.cs]: Remove unnecessary region

File: JsonBuilder/JsonBuilder/JsonBuilder.cs (hunk 2 above)
Comment (Darkstone2) [reply JsonBuilder/JsonBuilder/JsonBuilder.cs]: في كلاس جديد قصدك؟

File: JsonBuilder/JsonBuilder/JsonBuilder.cs [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+
+        /// <summary>
+        /// Add new property to json object
+        /// </summary>
+        /// <param name="property">property Name</param>
+        /// <param name="value">property value</param>
+        /// <returns></returns>
+        public JsonBuilder AppendProperty(string property, dynamic value)
+        {
+            if (property == null)
```
Comment (hishamco) [inline JsonBuilder/JsonBuilder/JsonBuilder.cs]:
```csharp
if(String.IsNullOrEmpty(property))
    throw ArgumentNullException("The JSON properties can't be null!");
```

File: JsonBuilder/JsonBuilder/JsonBuilder.cs [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+                throw new NullReferenceException("The JSON properties can't be null!");
+            property = property.Replace(" ", "_");
+
+            var expandoDict = this._currentObject as IDictionary<string, dynamic>;
```
Comment (hishamco) [inline JsonBuilder/JsonBuilder/JsonBuilder.cs]: No need for `this` qualifier

File: JsonBuilder/JsonBuilder/JsonBuilder.cs (hunk 4 above)
Comment (Darkstone2) [reply JsonBuilder/JsonBuilder/JsonBuilder.cs]: But what if the user put null to the JSON property!?

Comment (Darkstone2) [reply JsonBuilder/JsonBuilder/JsonBuilder.cs]: Or Enter a space that's not valid right?

File: JsonBuilder/JsonBuilder/JsonBuilder.cs [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+        /// <typeparam name="T">Class</typeparam>
+        /// <returns>True if the JSON object match the class, false otherwise</returns>
+        public bool IsMatchThisClass<T>() where T : class
+        {
+            var properties = typeof(T).GetProperties();
+            var expandoDict = this._currentObject as IDictionary<string, dynamic>;
+            return expandoDict.Keys.Count == properties.Length && properties.All(propertyInfo => expandoDict.Keys.Contains(propertyInfo.Name));
+        }
+
+        /// <summary>
```
Comment (hishamco) [inline JsonBuilder/JsonBuilder/JsonBuilder.cs]: Remove the comments for all `private` methods

File: JsonBuilder/JsonBuilder/JsonBuilder.cs (hunk 2 above)
Comment (Darkstone2) [reply JsonBuilder/JsonBuilder/JsonBuilder.cs]: Done

File: JsonBuilder/JsonBuilder/JsonBuilder.cs (hunk 1 above)
Comment (Darkstone2) [reply JsonBuilder/JsonBuilder/JsonBuilder.cs]: Done

File: JsonBuilder/JsonBuilder/JsonBuilder.cs (hunk 3 above)
Comment (Darkstone2) [reply JsonBuilder/JsonBuilder/JsonBuilder.cs]: Done

File: JsonBuilder/JsonBuilder/JsonBuilder.cs (hunk 5 above)
Comment (Darkstone2) [reply JsonBuilder/JsonBuilder/JsonBuilder.cs]: Done

File: JsonBuilder/JsonBuilder/JsonBuilder.cs (hunk 6 above)
Comment (Darkstone2) [reply JsonBuilder/JsonBuilder/JsonBuilder.cs]: Done

File: JsonBuilder/JsonBuilder/JsonBuilder.cs (hunk 4 above)
Comment (Darkstone2) [reply JsonBuilder/JsonBuilder/JsonBuilder.cs]: Ah, sorry got it!
Done

**Comment by hishamco** [comment]
@Darkstone2 push your changes, so I can merge

**Comment by Darkstone2** [comment]
@hishamco Ok done.

File: JsonBuilder/JsonBuilder/JsonBuilder.cs [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+
+            return false;
+        }
+
+        /// <summary>
+        /// Convert JSON string to .NET object
+        /// </summary>
+        /// <param name="jsonString">JSON string</param>
+        /// <returns>Dictionary Representing JSON Properties</returns>
+        private static dynamic ConvertToObject(string jsonString)
```
[the remark this answers is not in the document]
Comment (Darkstone2) [reply JsonBuilder/JsonBuilder/JsonBuilder.cs]: @hishamco Done

**Pull request merged**